File: backend/schemas/report.py
"""Automation report detail — PHASE_08."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class OwnershipRow(BaseModel):
    field: str
    value: str
    confidence: str = "low"
    source: str = ""


class ContactData(BaseModel):
    model_config = ConfigDict(extra="allow")

    name: str = ""
    title: str = ""
    company: str = ""
    email: str | None = None
    linkedin: str | None = None


class ColdEmailScript(BaseModel):
    subject: str = ""
    body: str = ""


class OutreachScripts(BaseModel):
    cold_email: ColdEmailScript = Field(default_factory=ColdEmailScript)
    linkedin: str = ""
    phone: str = ""


class PillarRationale(BaseModel):
    label: str
    score: float
    max_points: float
    detail: str


class WhyNowFactor(BaseModel):
    icon: str
    label: str
    points: float
    timing: str


class ScoreRationale(BaseModel):
    overall_line: str
    pillars: list[PillarRationale]
    why_now: list[WhyNowFactor]
    counterfactual_line: str


class AutomationReportDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    building_id: str
    building_name: str
    building_address: str
    score_at_trigger: float
    genome_archetype: str
    run_at: str
    threshold_at_trigger: int
    routed_to_rep_id: str | None

    ownership: list[OwnershipRow]
    contact: ContactData
    recent_news: str | None = None
    esg_commitments: str | None = None

    roof_sqft: int
    annual_gallons: float
    payback_years: float
    ct_detected: bool
    drought_label: str
    applicable_incentives: str

    outreach_scripts: OutreachScripts
    score_rationale: ScoreRationale

File: frontend/types/report.ts
export type OwnershipRow = {
  field: string;
  value: string;
  confidence: string;
  source: string;
};

export type ContactData = {
  name: string;
  title: string;
  company: string;
  email?: string | null;
  linkedin?: string | null;
};

export type AutomationReportDetail = {
  id: string;
  building_id: string;
  building_name: string;
  building_address: string;
  score_at_trigger: number;
  genome_archetype: string;
  run_at: string;
  threshold_at_trigger: number;
  routed_to_rep_id: string | null;
  ownership: OwnershipRow[];
  contact: ContactData;
  recent_news: string | null;
  esg_commitments: string | null;
  roof_sqft: number;
  annual_gallons: number;
  payback_years: number;
  ct_detected: boolean;
  drought_label: string;
  applicable_incentives: string;
  outreach_scripts: {
    cold_email: { subject: string; body: string };
    linkedin: string;
    phone: string;
  };
  score_rationale: {
    overall_line: string;
    pillars: { label: string; score: number; max_points: number; detail: string }[];
    why_now: { icon: string; label: string; points: number; timing: string }[];
    counterfactual_line: string;
  };
};
